elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchRules.java

```diff
@@ -113,8 +113,7 @@ static List<String> elasticsearchFieldNames(final RelDataType rowType) {
     return SqlValidatorUtil.uniquify(
         new AbstractList<String>() {
           @Override public String get(int index) {
-            final String name = rowType.getFieldList().get(index).getName();
-            return name.startsWith("$") ? "_" + name.substring(2) : name;
```

Comment:
I had test join operation of elasticsearch，it would return empty result. here name.substring(2) is not right which maybe copy from mongodb(mongodb is $_f1 and elasticsearch is $f1),and here not necessary to do this operation，return rowType.getFieldList().get(index).getName() directly is OK